trust-center-agent/src/test/java/care/smith/fts/tca/rest/DeIdentificationControllerTest.java

```diff
@@ -156,10 +159,32 @@ void researchMappingEmpty() {
   @Test
   void researchMappingWithAnyException() {
     given(mappingProvider.fetchResearchMapping("transferId"))
-        .willReturn(Mono.error(new TransferProcessException("")));
+        .willReturn(Mono.error(new TransferProcessException("error message")));
 
     create(controller.researchMapping("transferId"))
-        .expectError(TransferProcessException.class)
-        .verify();
+        .expectNext(
+            ResponseEntity.of(
+                    ProblemDetail.forStatusAndDetail(
+                        HttpStatus.INTERNAL_SERVER_ERROR, "error message"))
+                .build())
+        .verifyComplete();
+  }
+
+  @Test
+  void researchMappingInvalidDateShiftValueFromKeyValueStore() {
+    given(mappingProvider.fetchResearchMapping("transferId"))
+        .willReturn(
+            Mono.error(
+                new NumberFormatException(
+                    "Invalid dateShiftMillis value. For input string: \"nan\"")));
+
+    create(controller.researchMapping("transferId"))
+        .expectNext(
+            ResponseEntity.of(
+                    ProblemDetail.forStatusAndDetail(
+                        HttpStatus.INTERNAL_SERVER_ERROR,
+                        "Invalid dateShiftMillis value. For input string: \"nan\""))
```

Comment:
I would not return the invalid string `nan` to the client, because it could contain sensitive information. Please only log that string.